br/pkg/lightning/restore/precheck_impl.go

```diff
@@ -0,0 +1,1164 @@
+
+	switch {
+	case localAvailable > estimatedDataSizeWithIndex:
+		theResult.Message = fmt.Sprintf("local disk resources are rich, estimate sorted data size %s, local available is %s",
+			units.BytesSize(float64(estimatedDataSizeWithIndex)), units.BytesSize(float64(localAvailable)))
+		theResult.Passed = true
+	case int64(ci.cfg.TikvImporter.DiskQuota) > localAvailable:
+		theResult.Message = fmt.Sprintf("local disk space may not enough to finish import, estimate sorted data size is %s,"+
+			" but local available is %s, please set `tikv-importer.disk-quota` to a smaller value than %s"+
+			" or change `mydumper.sorted-kv-dir` to another disk with enough space to finish imports",
+			units.BytesSize(float64(estimatedDataSizeWithIndex)),
+			units.BytesSize(float64(localAvailable)), units.BytesSize(float64(localAvailable)))
+		theResult.Passed = false
+		log.FromContext(ctx).Error(theResult.Message)
+	default:
+		theResult.Message = fmt.Sprintf("local disk space may not enough to finish import, "+
+			"estimate sorted data size is %s, but local available is %s,"+
+			"we will use disk-quota (size: %s) to finish imports, which may slow down import",
+			units.BytesSize(float64(estimatedDataSizeWithIndex)),
+			units.BytesSize(float64(localAvailable)), units.BytesSize(float64(ci.cfg.TikvImporter.DiskQuota)))
+		theResult.Passed = true
+		log.FromContext(ctx).Warn(theResult.Message)
+	}
+	return theResult, nil
+}
+
+type checkpointCheckItem struct {
+	cfg           *config.Config
+	preInfoGetter PreRestoreInfoGetter
+	dbMetas       []*mydump.MDDatabaseMeta
+	checkpointsDB checkpoints.DB
+}
+
+func NewCheckpointCheckItem(cfg *config.Config, preInfoGetter PreRestoreInfoGetter, dbMetas []*mydump.MDDatabaseMeta, checkpointsDB checkpoints.DB) PrecheckItem {
+	return &checkpointCheckItem{
+		cfg:           cfg,
+		preInfoGetter: preInfoGetter,
+		dbMetas:       dbMetas,
+		checkpointsDB: checkpointsDB,
+	}
+}
+
+func (ci *checkpointCheckItem) GetCheckItemID() CheckItemID {
+	return CheckCheckpoints
+}
+
+func (ci *checkpointCheckItem) Check(ctx context.Context) (*CheckResult, error) {
+	if !ci.cfg.Checkpoint.Enable || ci.checkpointsDB == nil {
+		return nil, nil
+	}
+	theResult := &CheckResult{
+		Item:     ci.GetCheckItemID(),
+		Severity: Critical,
+		Passed:   true,
+		Message:  "the checkpoints are valid",
+	}
+
+	checkMsgs := []string{}
+	for _, dbInfo := range ci.dbMetas {
+		for _, tableInfo := range dbInfo.Tables {
+			msgs, err := ci.CheckpointIsValid(ctx, tableInfo)
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+			checkMsgs = append(checkMsgs, msgs...)
+		}
+	}
+	if len(checkMsgs) > 0 {
+		theResult.Passed = false
+		theResult.Message = strings.Join(checkMsgs, "\n")
+	}
+	return theResult, nil
+}
+
+// CheckpointIsValid checks whether we can start this import with this checkpoint.
+func (ci *checkpointCheckItem) CheckpointIsValid(ctx context.Context, tableInfo *mydump.MDTableMeta) ([]string, error) {
```

Comment:
Will this function be invoked outside this package? It seems it's only called by the `check`.